compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -621,16 +591,23 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                 {
                     // don't show type `_`
                     if span.desugaring_kind() == Some(DesugaringKind::ForLoop)
-                    && let ty::Adt(def, substs) = ty.kind()
-                    && Some(def.did()) == self.tcx.get_diagnostic_item(sym::Option)
+                        && let ty::Adt(def, substs) = ty.kind()
+                        && Some(def.did()) == self.tcx.get_diagnostic_item(sym::Option)
                     {
-                        err.span_label(span, format!("this is an iterator with items of type `{}`", substs.type_at(0)));
+                        err.span_label(
+                            span,
+                            format!(
+                                "this is an iterator with items of type `{}`",
+                                substs.type_at(0)
```

Comment:
why not a trailing comma here..

Reply:
rustfmt, probably. It usually ends up getting stuck inside of let chain if block.